src/components/tooltip/tooltip.tsx

```diff
@@ -0,0 +1,185 @@
+import * as debounce from 'debounce';
+import * as React from 'react';
+import {stylable} from 'wix-react-tools';
+
+import {GlobalEvent} from '../global-event';
+import {Portal} from '../portal';
+import styles from './tooltip.st.css';
+
+export type Position = 'top' | 'left' | 'right' | 'bottom';
+
+export interface TooltipProps {
+    children?: React.ReactNode;
+    position?: Position;
+    id: string;
+    open?: boolean;
+    showTrigger?: string;
+    hideTrigger?: string;
+    showDelay?: number;
+    hideDelay?: number;
+}
+
+export interface TooltipState {
+    style?: React.CSSProperties;
+    open: boolean;
+}
+
+@stylable(styles)
+class StyledTooltip extends React.Component<TooltipProps, TooltipState> {
+    public static defaultProps = {
+        open: false,
+        position: 'top',
+        showTrigger: 'mouseenter',
+        hideTrigger: 'mouseleave',
+        showDelay: 0,
+        hideDelay: 0
+    };
+
+    private target: HTMLElement | null = null;
+    private events: string[] = [];
+    private timeout?: number;
+    private onWindowResize = debounce(() => {
+        if (this.state.open) {
+            this.setStyles();
+        }
+    }, 500);
```

Comment:
this should be parametrized from props.

Reply:
no, I think, this should not be configurable. The only reason I add debounce is because `resize` fires to often